target/linux/qualcommbe/patches-6.6/901-03-arm64-dts-ipq9574-Update-the-IPQ9574-RDP433-DTS.patch

```diff
@@ -0,0 +1,507 @@
+From 5b4cb0d6d600942f5762c8f011582cd1af7f7546 Mon Sep 17 00:00:00 2001
+From: Luo Jie <quic_luoj@quicinc.com>
+Date: Thu, 22 May 2025 11:21:12 +0800
+Subject: [PATCH] arm64: dts: ipq9574: Update the IPQ9574 RDP433 DTS
+
+Update the PCS & PPE dts nodes to align with the updated drivers.
+
+Signed-off-by: Luo Jie <quic_luoj@quicinc.com>
+---
+ arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts | 148 +++++++++++++------
+ arch/arm64/boot/dts/qcom/ipq9574.dtsi       | 152 ++++++++------------
+ 2 files changed, 169 insertions(+), 131 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts b/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts
+index 6bebd57f7..427829b78 100644
+--- a/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts
++++ b/arch/arm64/boot/dts/qcom/ipq9574-rdp433.dts
+@@ -69,6 +69,7 @@ ethernet-phy-package@0 {
+ 
+ 		phy0: ethernet-phy@10 {
+ 			reg = <0x10>;
++			clocks = <&cmn_pll ETH_25MHZ_CLK>;
+ 		};
+ 
+ 		phy1: ethernet-phy@11 {
+@@ -86,6 +87,7 @@ phy3: ethernet-phy@13 {
+ 
+ 	phy4: ethernet-phy@8 {
+ 		compatible ="ethernet-phy-ieee802.3-c45";
++		clocks = <&cmn_pll ETH1_50MHZ_CLK>;
+ 		reg = <8>;
+ 
+ 		nvmem-cells = <&aqr_fw>;
+@@ -94,6 +96,7 @@ phy4: ethernet-phy@8 {
+ 
+ 	phy5: ethernet-phy@0 {
+ 		compatible ="ethernet-phy-ieee802.3-c45";
++		clocks = <&cmn_pll ETH2_50MHZ_CLK>;
+ 		reg = <0>;
+ 
+ 		nvmem-cells = <&aqr_fw>;
+@@ -220,6 +223,67 @@ rclk-pins {
+ 	};
+ };
+ 
++&pcs0 {
++        status = "okay";
++};
++
++&pcs0_ch0 {
++        clocks = <&nsscc NSS_CC_UNIPHY_PORT1_RX_CLK>,
++                 <&nsscc NSS_CC_UNIPHY_PORT1_TX_CLK>;
++        clock-names = "rx",
++                      "tx";
++        status = "okay";
++};
++
++&pcs0_ch1 {
++        clocks = <&nsscc NSS_CC_UNIPHY_PORT2_RX_CLK>,
++                 <&nsscc NSS_CC_UNIPHY_PORT2_TX_CLK>;
++        clock-names = "rx",
++                      "tx";
++        status = "okay";
++};
++
++&pcs0_ch2 {
++        clocks = <&nsscc NSS_CC_UNIPHY_PORT3_RX_CLK>,
++                 <&nsscc NSS_CC_UNIPHY_PORT3_TX_CLK>;
++        clock-names = "rx",
++                      "tx";
++        status = "okay";
++};
++
++&pcs0_ch3 {
++        clocks = <&nsscc NSS_CC_UNIPHY_PORT4_RX_CLK>,
++                 <&nsscc NSS_CC_UNIPHY_PORT4_TX_CLK>;
++        clock-names = "rx",
++                      "tx";
++        status = "okay";
++};
```

Comment:
The PCS MIIs, PCS, and NSSCC clock controller are part of the SOC, right? So the clocks are routed internally in the SOC, and are not board-dependent, right?

So why are these moved from the SOC dtsi to the board DTS? Seems like every board now needs to duplicate the same PCS MII clock routing.

Reply:
Shouldn't then the `clocks` and `clock-names` properties be part of the DTSI?